### DISCUSSION TITLE
Performance of using REVERSELY

### DISCUSSION THREAD

**Discussion by PhungQuocLuong-git** [vesoft-inc/nebula#5324] [Q&A]
Hello, 
I am analysing graph 2-way-relationship presented by 2 edges( Ex: Sam has a pen/ A pen is belonged to Sam). I wonder if I use REVERSELY which can optimize my relationship down to 1 edge, does it decrease graph performance compared to using 2 edges presented for that 2-way-relationship ?
Thank :))

**Comment by QingZ11** [comment]
In nebulagraph, although you have inserted an `a->b` edge, you can use the `REVERSELY` clause to find the reverse edge. refer docs:[https://docs.nebula-graph.io/3.4.0/3.ngql-guide/7.general-query-statements/3.go/#syntax](https://docs.nebula-graph.io/3.4.0/3.ngql-guide/7.general-query-statements/3.go/#syntax)

In a word, any edge in the nebulagraph is stored in two copies, the outgoing edge and the incoming edge
